src/safeposix/syscalls/fs_calls.rs

```diff
@@ -2180,31 +2223,75 @@ impl Cage {
             }
             Some(_) => {
                 return syscall_error(
-                    Errno::EACCES,
+                    Errno::ENOTDIR,
                     "fchdir",
-                    "cannot change working directory on this file descriptor",
+                    "the file descriptor does not refer to a directory",
                 )
             }
             None => return syscall_error(Errno::EBADF, "fchdir", "invalid file descriptor"),
         };
-
+        //Obtain the write lock on the current working directory of the cage
+        //and change it to the new directory
         let mut cwd_container = self.cwd.write();
-
-        *cwd_container = interface::RustRfc::new(convpath(path_string.as_str()));
+        *cwd_container = interface::RustRfc::new(normpath(convpath(path_string.as_str()), self));
```

Comment:
Why do we need to call normpath as well as convpath?

Reply:
covpath just converts the string into a rust path object, while normpath gets rid of "." and ".." references while also removing the unnecessary "/" at the end of the path name